provider/defaultingress/ingress.go

```diff
@@ -425,31 +425,34 @@ func (r *DefaultIngressResource) updateIngress(ctx context.Context, state, plan
 	return nil
 }
 
-func getDefaultIngressBuilder(ctx context.Context, state *DefaultIngress) *cmv1.IngressBuilder {
+func getDefaultIngressBuilder(ctx context.Context, state, plan *DefaultIngress) *cmv1.IngressBuilder {
 	ingressBuilder := cmv1.NewIngress()
-	routeSelectors, err := common.OptionalMap(ctx, state.RouteSelectors)
-	if err != nil {
-		return nil
+	if !reflect.DeepEqual(state.RouteSelectors, plan.RouteSelectors) {
+		routeSelectors, err := common.OptionalMap(ctx, plan.RouteSelectors)
+		if err != nil {
+			return nil
+		}
+		if routeSelectors == nil {
```

Comment:
why can't we sent the nil as is (in case of nil list we sending it as a nil object)

Reply:
If we send nil the SDK does not set the value in the request. As it omits nil values, so then the API will not pick this up

Reply:
This was already being sent as empty, but there's was no check that the plan differed from the current state

Reply:
Why is it needed to be validated?
In case there is no change that updating the same value should do nothing, right?

Reply:
Depending on the version and also some conditions there are values that cannot be supplied. By setting this to some value without comparison it ends up trying to supply when it should be nil and API responds with error